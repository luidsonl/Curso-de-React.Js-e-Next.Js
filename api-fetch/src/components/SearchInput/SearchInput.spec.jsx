import { render, screen } from '@testing-library/react'
import { SearchInput } from '.'
import userEvent from '@testing-library/user-event'

describe('<SearchInput/>', ()=>{
    it('Should have a value of searchValue',()=>{
        const fn = jest.fn()
        render(<SearchInput handleChange={fn} searchValue={'Teste'}/>)

        const input = screen.getByPlaceholderText(/digite sua busca/i)
        
        expect(input.value).toBe('Teste')
    })

    it('Should call handlechange function on update',()=>{
        const fn = jest.fn()
        render(<SearchInput handleChange={fn} />)

        const input = screen.getByPlaceholderText(/digite sua busca/i)

        const value = 'valor'

        userEvent.type(input, value)

        expect(fn).toBeCalledTimes(value.length)
        
    })

    it('Should shoud match snapshot',()=>{
        const fn = jest.fn()
        const {container} = render(<SearchInput handleChange={fn} />)

        expect(container).toMatchSnapshot()
        
    })

})