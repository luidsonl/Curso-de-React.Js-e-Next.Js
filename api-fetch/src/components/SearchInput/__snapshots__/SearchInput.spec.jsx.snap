// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput/> Should shoud match snapshot 1`] = `
<div>
  <div
    class="search-container"
  >
    <div
      class="search-title"
    >
      Buscar
    </div>
    <input
      class="button-container"
      placeholder="Digite sua busca"
      type="search"
      value=""
    />
  </div>
</div>
`;
